### PR TITLE
WIP Initial files for Kubernetes doc

### PR DESCRIPTION
**Pull request by gtardif**
Signed-off-by: Guillaume Tardif <guillaume.tardif@docker.com>

### Proposed changes

Draft for Kubernetes integration in Docker for Mac 
(to be released first in Docker for Mac Edge 17.12, restricted to selected beta users). 

cc @mistyhacks I created this PR on the vnext-engine branch, not sure this is the right. Also, it will need to be integrated with the rest of the docs. 
And there will certainly be other modifications before this can be deployed.

### CODE REVIEW

**Comment by gtardif** [comment]
Current build with the latest Gui changes : https://download-stage.docker.com/mac/master/20866/Docker.dmg

File: docker-for-mac/kubernetes-integration.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Docker for Mac has a beta program that includes a full Kubernetes cluster, starting from the Edge 17.12 release.
+To be enrolled in the beta program, you can register [here](https://beta.docker.com/)
+
+## Activating Kubernetes
+
+If you have been enrolled in the beta program, once logged in with your dockerid, you will see a new "Kubernetes" tab in the preference pane : 
+
+![Docker kubernetes tab](images/enable-kubernetes.png)
+
+When activating Kubernets for the first time, it will download required kubernetes images (this might take some time). A single-node Kubernetes cluster is then setup and started.
```
Comment (gbarr01) [inline docker-for-mac/kubernetes-integration.md]: set up

File: docker-for-mac/kubernetes-integration.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- /mackit/kubernetes/
+title: Docker for Mac Kubernetes integration
+---
+
+Docker for Mac has a beta program that includes a full Kubernetes cluster, starting from the Edge 17.12 release.
+To be enrolled in the beta program, you can register [here](https://beta.docker.com/)
+
+## Activating Kubernetes
+
+If you have been enrolled in the beta program, once logged in with your dockerid, you will see a new "Kubernetes" tab in the preference pane :
```
Comment (gbarr01) [inline docker-for-mac/kubernetes-integration.md]: pane:

File: docker-for-mac/kubernetes-integration.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+You can check the Kubernetes cluster is ready : 
+``` 
+$ kubectl get nodes
+NAME                 STATUS    ROLES     AGE       VERSION
+docker-for-desktop   Ready     master    23h       v1.8.2
+```
+
+## Starting containers on the kubernetes clustrer
+
+Once the kubernetes cluster is started, you can start running usual docker commands to deploy services, they will get deployed in the Kubernetes cluster.
```
Comment (gbarr01) [inline docker-for-mac/kubernetes-integration.md]: you can start running the usual docker commands to deploy services.

Comment (gbarr01) [reply docker-for-mac/kubernetes-integration.md]: and maybe even: 
you can start running the usual _docker_ commands to deploy services.

File: docker-for-mac/kubernetes-integration.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+You can deploy services with `docker stack deploy --compose-file xxx`.  
+- `kubectl get services` will show the services started.
+
+With Kubernetes you can also use namespaces. If not specified, services get deployed in the default namespace in Kubernetes.
```
Comment (gbarr01) [inline docker-for-mac/kubernetes-integration.md]: `default` namespace

**Comment by gtardif** [comment]
@gbarr01 Actually I started this PR at first when I didn't know exactly how we were going to move with this doc. After our demo on Monday I thought you were going to update/finalize this with screenshots etc. 
It might be easier if you start another PR and amend directly on it ? (& grab what you need from this one).

**Comment by mistyhacks** [comment]
I'll get a PR up for this today. Looks like it needs to land Wednesday? Please slack me if that's not right.

File: docker-for-mac/kubernetes-integration.md (hunk 2 above)
Comment (mistyhacks) [reply docker-for-mac/kubernetes-integration.md]: You don't actually need to be logged in.

**Comment by mistyhacks** [comment]
Closed in favor of #5439

**Pull request closed without merging**